test/performance/alignment/global_affine_alignment_simd_benchmark.cpp

```diff
@@ -0,0 +1,185 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+#include <benchmark/benchmark.h>
+
+#include <algorithm>
+#include <iostream>
+#include <memory>
+#include <random>
+#include <utility>
+#include <vector>
+
+#include <seqan3/alignment/pairwise/align_pairwise.hpp>
+#include <seqan3/alphabet/aminoacid/aa20.hpp>
+#include <seqan3/alphabet/nucleotide/dna4.hpp>
+#include <seqan3/range/views/to.hpp>
+#include <seqan3/range/views/zip.hpp>
+#include <seqan3/test/performance/units.hpp>
+#include <seqan3/test/seqan2.hpp>
+#include <seqan3/std/ranges>
+
+#include <seqan3/core/debug_stream.hpp>
+
+#ifdef SEQAN3_HAS_SEQAN2
+    #include <seqan/align.h>
+    #include <seqan/align_parallel.h>
+#endif
+
+#include "test_sequence_generation.hpp" // Shared functionality to generate test sequences.
+
+using namespace seqan3;
+using namespace seqan3::test;
+
+constexpr auto affine_cfg = align_cfg::mode{global_alignment} |
+                            align_cfg::gap{gap_scheme{gap_score{-1}, gap_open_score{-10}}} |
+                            align_cfg::scoring{nucleotide_scoring_scheme{match_score{4}, mismatch_score{-5}}};
+
+// Aliases to beautify the benchmark output
+using score = detail::with_score_type;
+using end_position = detail::with_back_coordinate_type;
+using trace = detail::with_alignment_type;
+
+// Globally defined constants to ensure same test data.
+inline constexpr size_t sequence_length = 150;
+inline constexpr size_t set_size        = 1024;
+
+// Range to test for sequence length variance
+inline constexpr size_t variance_begin = 0;
+inline constexpr size_t variance_end = 64;
+inline constexpr size_t variance_step = 8;
+
+// ============================================================================
+//  affine; score; dna4; collection
+// ============================================================================
+
+template <typename result_t>
+void seqan3_affine_dna4_simd(benchmark::State & state)
+{
+    size_t sequence_length_variance = state.range(0);
+    auto [vec1, vec2] = generate_data_seqan3<seqan3::dna4>(sequence_length, sequence_length_variance, set_size);
+
+    auto data = views::zip(vec1, vec2) | views::to<std::vector>;
```

Comment:
The function `generate_sequence_pairs` would return a vector of pairs right away...

Reply:
Yes, that would fit. I would still propose to fix this altogether with the changes necessary for passing one random generator per test instance to the generator function.